arch/arm/core/cortex_m/tz/CMakeLists.txt

```diff
@@ -25,11 +25,20 @@ if(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
     ${IMAGE}extra_post_build_byproducts
     ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
     )
+
+  # Add wrapper target for the benefit of make.
+  add_custom_target(nsc_veneers
+    DEPENDS
+    ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
+    ${ZEPHYR_FINAL_EXECUTABLE})
 endif()
 
-# Link the entry veneers library file with the Non-Secure Firmware that needs it.
-zephyr_link_libraries_ifdef(CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS
-  ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
-  )
+if (DEFINED CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS)
+  # Link the entry veneers library file with the Non-Secure Firmware that needs it.
+  zephyr_link_libraries(${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME})
```

Comment:
> ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}

We are repeating this three times now, please create a variable for it.

Reply:
Done